### DISCUSSION TITLE
2021-10-13 Node Experience Team Update

### DISCUSSION THREAD

**Discussion by janewang** [near/community#112] [Node]
Node Experience team project board: https://github.com/orgs/near/projects/31
Attended: @mina86 @nikurt @janewang @bowenwang1996 @posvyatokum @marcelo-gonzalez

# Done
- @mina86 added monitoring and basic alerting to NayDuck. near/nayduck#11
- @posvyatokum added saving TX hashes and env in runtime tester to check TX status after execution near/nearcore#4374
- @posvyatokum fixed contract deployment in fuzzing near/nearcore#4374
- @nikurt submitted PR (https://github.com/near/nearcore/pull/4867) for faster replaying the chain history, and better handling of missing chunks. This PR even has two tests. near/nearcore#4597
- @janewang and @bowenwang1996 hosted the October NVAB meeting and will be creating a process for electing a board chair

# In Progress
- @mina86 added regression detection to NayDuck though formatting of the alert and rules for routing might still need adjustments. near/nayduck#4
- @mina86 added a CI test to check whether it’s possible to spin a simple cluster up.  The code is done and ready and is blocked on fix for the current brokenness.  near/nearcore#4993
- @mina86 added a CI test verifying that the protocol version does not increase by more than one during release.  There are still corner cases to handle but the bulk of cases where this issue can arise is not checked.   near/nearcore#4956
- @mina86 continues investigating indexer going out of sync.  Currently progress is slow.  near/nearcore#4937
- @mina86 continues investigating the performance cliff in one of the tests.  The problem appears to be somewhere in RocksDB (perhaps the way it handles caches?) near/nearcore#4771
- @posvyatokum Working on fuzzing near/nearcore#4374:
-- adding setting max prepaid gas limit so expensive txs will actually be executing
-- adding txs with data receipts in fuzzing
- @posvyatokum Generally working on fully covering runtime/runtime/src/lib.rs with fuzzing near/nearcore#4374
- @posvyatokum Checked nayduck fuzzing fails, increased time limit for block production  near/nearcore#4982
- @marcelo-gonzalez is working on adding more prometheus metrics near/nearcore#2834
- @nikurt runs multiple mocknet tests for the upcoming simple sharding protocol upgrade.
- @nikurt helped with the recent betanet incident and documented it [Betanet incident - 2021-10-08 - Simple Sharding incompatibility ](https://docs.google.com/document/d/1l4Qq9P-0kXz2tfm_XWafLobaB-YYPbydsb-mcZpnd-M/edit)

# Meta
All issues for Q4 OKRs are created and added to the [Q4 NodeX board](https://github.com/orgs/near/projects/31)

# Upcoming
- The next testnet release for nearcore 1.22.0 is Monday, October 18 ⚙️
- The next mainnet release for nearcore 1.22.0 is Monday, November 15 ⚙️

**Comment by maxzaver** [comment]
> This PR even has two tests

Lol
